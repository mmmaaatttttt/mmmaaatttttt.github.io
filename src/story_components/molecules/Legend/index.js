import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { FlexContainer, Icon } from "story_components";
import COLORS from "utils/styles";

const StyledHeader = styled.h4`
  margin-bottom: 0.1rem;
`;

const StyledLabelSpan = styled.span`
  margin-left: 0.25rem;
`;

const StyledLabelWrapper = styled.div`
  padding: 0 0.5rem;
`;
const defaultLabels = [
  {
    color: COLORS.BLUE,
    text: "Blue label"
  },
  {
    color: COLORS.RED,
    text: "Red label"
  }
];

function Legend({ title = "Default Title", labels = defaultLabels }) {
  return (
    <FlexContainer column cross="center" main="center">
      <StyledHeader>{title}</StyledHeader>
      <FlexContainer main="center" width="100%" shouldWrap>
        {labels.map(({ color, text }) => (
          <StyledLabelWrapper key={color}>
            <Icon color={color} name="square" />
            <StyledLabelSpan>{text}</StyledLabelSpan>
          </StyledLabelWrapper>
        ))}
      </FlexContainer>
    </FlexContainer>
  );
}

Legend.propTypes = {
  title: PropTypes.string,
  labels: PropTypes.arrayOf(
    PropTypes.shape({
      color: PropTypes.string,
      text: PropTypes.string
    })
  )
};

export default React.memo(Legend);
