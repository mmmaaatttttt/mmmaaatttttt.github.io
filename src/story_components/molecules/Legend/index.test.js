import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Legend from ".";

it("renders successfully", () => {
  shallow(<Legend />);
});

it("matches snapshot with props passed in", () => {
  const wrapper = shallow(<Legend />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
