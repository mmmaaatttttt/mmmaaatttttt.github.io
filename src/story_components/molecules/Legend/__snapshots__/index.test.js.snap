// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props passed in 1`] = `
<styled.div
  column={true}
  cross="center"
  flex={1}
  main="center"
  margin="0"
  shouldWrap={false}
  textAlign="left"
  width="auto"
>
  <styled.h4>
    Default Title
  </styled.h4>
  <styled.div
    column={false}
    cross="stretch"
    flex={1}
    main="space-between"
    margin="0"
    shouldWrap={true}
    textAlign="left"
    width="100%"
  >
    <div
      key="#5ECFFF"
    >
      <Icon
        color="#5ECFFF"
        disabled={false}
        hover={false}
        name="square"
        opacity={1}
        size={1}
        type="solid"
      />
      <styled.span>
        Blue label
      </styled.span>
    </div>
    <div
      key="#FF3C23"
    >
      <Icon
        color="#FF3C23"
        disabled={false}
        hover={false}
        name="square"
        opacity={1}
        size={1}
        type="solid"
      />
      <styled.span>
        Red label
      </styled.span>
    </div>
  </styled.div>
</styled.div>
`;
